src/strat_engine/pool.rs

```diff
@@ -0,0 +1,111 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::collections::BTreeMap;
+use std::vec::Vec;
+use std::path::Path;
+
+use uuid::Uuid;
+
+use engine::EngineResult;
+use engine::Pool;
+use engine::Filesystem;
+use engine::Dev;
+use engine::Cache;
+
+use super::blockdev::BlockDev;
+use super::consts::*;
+
+#[derive(Debug, Clone)]
+pub struct StratFilesystem {
+    pub name: String,
+    pub thin_id: u32,
+}
+
+#[derive(Debug)]
+pub struct StratPool {
+    pub name: String,
```

Comment:
That seems fine. But it makes the problem only one field smaller. And a line between the identifying value and the rest would be helpful. But I'm also happy to leave this until a (somewhat) later resolution.